pkg/quotaplugins/quota-forest/quota-manager/quota/quotamanager.go

```diff
@@ -372,9 +376,36 @@ func (m *Manager) AllocateForest(forestName string, consumerID string) (response
 	if consumerInfo == nil {
 		return nil, fmt.Errorf("consumer %s does not exist, create and add first", consumerID)
 	}
-	if forestController.IsConsumerAllocated(consumerID) {
-		return nil, fmt.Errorf("consumer %s already allocated on forest %s", consumerID, forestName)
+	//allocation can pass if consumers inside the forest are not released
+	//make sure that appwrapper (consumer) does not exist in target namespace
+	namespace, awName := util.ParseId(consumerID)
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		klog.Errorf("unable to retrieve in-cluster kubeconfig %v", err)
+	}
+	arbclients := clientset.NewForConfigOrDie(config)
```

Comment:
I think that the arbclient should be at minimum an instance variable of the `Manager` struct. It makes no seense to instantiate the client at this point (if we even need the client at all)